File: gauntlet/entries/regex-find-all-4mb/go/main.go
package main

import (
	"bytes"
	"fmt"
	"os"
)

func main() {
	path := "corpus.txt"
	if len(os.Args) > 1 {
		path = os.Args[1]
	}
	text, err := os.ReadFile(path)
	if err != nil {
		panic(err)
	}
	fmt.Printf("matches %d\n", bytes.Count(text, []byte("struct")))
}

File: gauntlet/entries/regex-find-all-4mb/rust/main.rs
use std::env;
use std::fs;
use std::io;

fn main() -> io::Result<()> {
    let path = env::args().nth(1).expect("missing input");
    let text = fs::read_to_string(path)?;
    let matches = text.match_indices("struct").count();
    println!("matches {matches}");
    Ok(())
}
